src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1156,6 +1159,47 @@ class WorkspaceService(protected val userInfo: UserInfo, dataSource: DataSource,
     }
   }
 
+
+  //todo: clean this up, also accept a user and check if they can write to bucket, if they're in proxy, and if their proxy is in appropriate group
+  def getWorkspaceStatus(workspaceName: WorkspaceName, userSubjectId: RawlsUserSubjectId): Future[PerRequestMessage] = {
+    asAdmin {
+      dataSource.inFutureTransaction(readLocks = Set(workspaceName)) { txn =>
+        withWorkspaceContext(workspaceName, txn) { workspaceContext =>
+          val user = containerDAO.authDAO.loadUser(RawlsUserRef(userSubjectId), txn).get
+          val maxessLevel = containerDAO.authDAO.getMaximumAccessLevel(user, workspaceContext.workspace.workspaceId, txn)
+          val bucketName = workspaceContext.workspace.bucketName
+          val accessLevels = WorkspaceAccessLevels.groupAccessLevelsAscending
+
+          val rawlsGroupStatuses = accessLevels.map { lvl =>
+            s"RAWLS_${lvl}_EXISTS" -> !containerDAO.authDAO.loadGroup(RawlsGroupRef(RawlsGroupName(UserAuth.toWorkspaceAccessGroupName(workspaceName, lvl))), txn).isEmpty
+          }
+
+          val googleGroupStatuses = accessLevels.map { lvl =>
+            toFutureTry(gcsDAO.getGoogleGroup(gcsDAO.toGroupId(bucketName, lvl)).map(s"GOOGLE_${lvl}_EXISTS" -> _.isDefined))
+          }
+
+          val googleBucketStatuses = {
+            Seq(toFutureTry(gcsDAO.getBucket(bucketName).map("GOOGLE_BUCKET_EXISTS" -> _.isDefined)),
+              toFutureTry(gcsDAO.getBucketACL(bucketName).map("GOOGLE_BUCKET_ACL_VALID" -> _.get.size.equals(5))))
```

Comment:
there's unfortunately no way to get a bucket ACL and see the actual emails stored for each entry. it looks like the best we can do is check if there are the correct number of entries in the bucket ACL